scripts/db/src/get-user.ts

```diff
@@ -23,51 +25,91 @@ async function getByEmail(email: string, callback: DbScriptCallback) {
+        .collection<ForumUser>('users')
+        .findOne({ 'emails.address': email })
+
+      await mongoClient.close()
+
+      if (!forumUser) {
+        return null
+      }
+      return {
+        id: forumUser._id,
+        nickname: forumUser.displayName,
+        email: forumUser.email,
+      }
     }
 
-    /** Construct a postgres client and connect to the server */
-    const client: Client = new Client(connectionInfo)
-    await client.connect()
+    async function getParfitUser(): Promise<CallbackUser | null> {
+      /** Declare connection info */
+      const pgConnectionInfo: PGConnectionConfig = {
+        user: configuration.POSTGRES_USERNAME,
+        password: configuration.POSTGRES_PASSWORD,
+        host: configuration.POSTGRES_HOST,
+        database: configuration.POSTGRES_DATABASE,
+        port: configuration.POSTGRES_PORT
+          ? parseInt(configuration.POSTGRES_PORT)
+          : 5432,
+        ssl: TEMPLATE_DATA.pgShouldSsl,
+      }
+      /** Construct a postgres client and connect to the server */
+      const pgClient: PGClient = new PGClient(pgConnectionInfo)
+      await pgClient.connect()
 
-    /** Get the person based on their email, joining on the password table */
-    const query = `
-      select
-        person.id, email, first_name, last_name, password
-      from people.person
-      join auth.password on password.person_id = person.id
-      where person.email = $1
-    `
-    const result = await client.query<PersonResult>(query, [email])
+      /** Get the person based on their email, joining on the password table */
+      const parfitQuery = `
+        select
+          person.id, email, first_name, last_name, password
+        from people.person
+        join auth.password on password.person_id = person.id
+        where person.email = $1
+      `
+      const parfitResult = await pgClient.query<PersonResult>(parfitQuery, [
+        email,
+      ])
 
-    /** Close the connection */
-    await client.end()
+      /** Close the connection */
+      await pgClient.end()
 
-    if (result.rows.length === 0) {
-      return callback(null)
-    }
+      const Person = parfitResult.rows[0]
 
-    const Person = result.rows[0]
+      if (parfitResult.rows.length === 0) {
+        return null
+      }
+      return {
+        id: Person.id,
+        given_name: Person.first_name,
+        family_name: Person.last_name,
+        email: Person.email,
+      }
+    }
 
-    /** Return the valid user back to Auth0 */
-    return callback(null, {
-      id: Person.id,
-      given_name: Person.first_name,
-      family_name: Person.last_name,
-      email: Person.email,
-    })
+    /** Give priority to Forum users, as the integration is newer */
+    const forumUser = await getForumUser()
+    if (forumUser) {
+      return callback(null, forumUser)
+    }
+    const parfitUser = await getParfitUser()
+    if (parfitUser) {
+      return callback(null, parfitUser)
+    }
+    return callback(null)
```

Comment:
Is this right? We previously didn't call the callback with an empty user, right?

Reply:
It's an empty error message and an undefined rule, and it's what we previously did.